### ISSUE TITLE
Switch behavior breaks backward functionality (iOS)

### ISSUE THREAD

**Issue by pfaucon** [xamarin/Xamarin.Forms#4049]
The old mechanism for setting switch cell "onColor" was:
UISwitch.Appearance.OnTintColor = new UIColor(1.0f, 1.0f, 1.0f, 1.0f);

The new mechanism involves setting the OnColor in the Xamarin.Forms Switch object. However, if you do not set an OnColor the old mechanism stops working.  This behavior is due to the way the [renderer](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/Renderers/SwitchRenderer.cs) is written.  In specific:
```
void UpdateOnColor()
{
	if (Element != null)
	{
		if (Element.OnColor == Color.Default)
			Control.OnTintColor = _defaultOnColor;
		else
			Control.OnTintColor = Element.OnColor.ToUIColor();
	}
}
```

This will always override the default value.  I suggest changing it so that the renderer stores a copy fo the last color that was set (initialized to Default), if the color changes to the same thing (e.g. Default -> Default) nothing happens, otherwise the color is set.

As an aside, this change was not copied over into the SwitchCell, so to maintain the old functionality one currently needs both the old fix, and the new one.

**Comment by samhouts** [comment]
related to #4027

**Comment by hartez** [comment]
This is a duplicate of #3111 and is fixed by #3819. The fix is available in the current prerelease and will be in the next stable release.
